### DISCUSSION TITLE
[Technical Questions]

### DISCUSSION THREAD

**Discussion by lowbaj21** [CS342/.github#16] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

Unable to add library with my views to Xcode project

### Reproduction

I went to CS342Application and selected add frameworks or libraries. There was no option that include the folder I added in Xcode, so I selected Add Other>Add Files and selected my folder CS342MemberViews>Sources>Alexis. However, the library still did not add. I've been following along with the lecture video and this issue didn't arise there, so I'm not sure how to fix this. I've already added my target information in the Package file.

<img width="255" alt="image" src="https://user-images.githubusercontent.com/71662608/214397661-92d944e0-d1c0-4b3b-a21d-300769ae654d.png">

<img width="403" alt="image" src="https://user-images.githubusercontent.com/71662608/214397714-f053ca6b-03a5-4ed9-a568-d554bd17c0e5.png">

 ```
   ]
        ),
        .target(
            name: "AlexisLowber",
            dependencies: [
                .target(name: "InstructorViews")
            ],
            exclude: [
                "Resources/AlexisLowber.jpeg.license",
                "Resources/AlexisLowberBio.md.license"
            ],
            resources: [
                .process("Resources")
            ]
        ),
        
```

### Expected behavior

I'm able to add my library and build without errors

### Additional context

I'm currently getting 4 errors:

Missing package product 'OliverAalami'
- Package resolution errors must be fixed before building
Missing package product 'PaulSchiedmayer'
- Package resolution errors must be fixed before building
Missing package product 'RajPabari'
- Package resolution errors must be fixed before building
Missing package product 'AshleyGriffin'
- Package resolution errors must be fixed before building

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment]
Hi Alexis,

I would start by following these steps to try and clear out the issues you're seeing:

- Clean all issues using Product > Clear All issues.
- Rebuild all files by choosing Product > Clean Build Folder (⇧⌘K).
- Reset the Xcode Package Cache using Packages > Reset Package Caches.
- Update all packages using Packages > Update to Latest Package Versions.
- Then try to run the build again: Product > Build (⌘B).

Let us know if this helps.

**Reply by lowbaj21** [reply]
Hi! 

I followed the steps, but I'm still getting the same errors unfortunately. Anything else I should try?

**Reply by vishnuravi** [reply]
Hey @lowbaj21, could you push your code to a branch on the CS342-2023-Application repository and then let us know so we can take a look?

**Reply by lowbaj21** [reply]
Sure, just pushed it. I recreated the project and tried starting over, but still can't seem to resolve the problems. My branch is [features/alexis](https://github.com/CS342/CS342-2023-Application/tree/feature/alexis) Thanks!

**Comment by PSchmiedmayer** [comment, marked as answer]
@lowbaj21,

Thank you for posting your question!

I have taken a look at your problem and your current state at [feature/alexis](https://github.com/CS342/CS342-2023-Application/tree/feature/alexis).
You get the following error message when asking Xcode to build your project:
```
public headers ("include") directory path for 'AlexisLowber' is invalid or not contained in the target
```

While it is generally not an easy error message to understand, it is most often related to the fact that the Swift Package Manager can not create a public interface for your target.
In your case, this is related to the fact that you do not have a Swift file in your target. You only have a few resources in the _Resources_ folder as well as the `AlexisLowberView.swift` file in the Resources folder. The Swift Package Manager handles all files in the _Resources_ folder as resources and does not feed them to the compiler as you have defined the _Resources_ folder as a special folder containing these resources in your target definition:
```swift
.target(
    name: "AlexisLowber",
    dependencies: [
        .target(name: "InstructorViews")
    ],
    exclude: [
        "Resources/AlexisLowber.jpeg.license",
        "Resources/AlexisLowber.md.license"
    ],
    resources: [
        .process("Resources")
    ]
)
```

You can therefore fix your error if you move the `AlexisLowberView.swift` file one folder up into the _AlexisLowber_ folder.

To make the view accessible outside of the package, you will then have to declare it public using the `public` access modifier, e.g., (I have removed the `body` of the view to make the code example simpler to read:
```swift
public struct AlexisLowber: View {
    public var body: some View {
        // ...
    }
    
    
    public init() { }
}
```

In general also advise going through these steps here to resolve to build errors in Xcode: https://github.com/orgs/CS342/discussions/6#discussioncomment-4742407 . If you run into any other issues we would recommend going through these steps.

Let us know if this resolves your issue, and then mark this message as an answer or let us know about any follow-up questions!

**Reply by lowbaj21** [reply]
This fixed it - thanks for the help!

**Reply by PSchmiedmayer** [reply]
@lowbaj21 Great to hear 🥳🚀
